### PR TITLE
[test_route_flap] Route flap multi-dut fix if the selected duthost has downstream neighbors

### PR DESCRIPTION
**Pull request by sanjair-git**
### Description of PR

Summary:
Fixes # (issue)

-  This PR fixes issue #11460 

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205
- [x] 202305
- [x] 202311

#### What is the motivation for this PR?

- For T2 Chassis(topo-t2), route/test_route_flap works fine, if the selected duthost has upstream T3 neighbors.
- Similarly, if the selected duthost has downstream T1 neighbors and if there is only one dut with T1 neighbors, test works fine too.
- But if the selected duthost has downstream T1 neighbors and if there are more than one dut with downstream neighbors, route/test_route_flap fails.

#### How did you do it?

- Fetch the 'neighbor type' (_LeafRouter/SpineRouter/AZNGhub_) based on the chosen _dev_port_ from _get_dev_port_and_route_
- Get all the neighbors of same neighbor type and fetch the ptf receiving ports list based on the retrieved neighbors list.
- Pass the received ptf ports list to send_recv_ping_packet method for further testing. 

#### How did you verify/test it?

- Ran _route/test_route_flap_ test case on T2 Chassis with 'topo-t2' topology having 3-line cards with at least 2 cards having downstream T1 neighbors.
-  Test passes.

### Documentation

image

### CODE REVIEW

File: tests/route/test_route_flap.py [hunk 1]
```diff
@@ -98,6 +98,96 @@ def get_ptf_recv_ports(duthost, tbinfo):
...
+    neighbor_type = ''
+    config_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    neighs = config_facts['BGP_NEIGHBOR']
+    dev_neigh_mdata = config_facts['DEVICE_NEIGHBOR_METADATA'] if 'DEVICE_NEIGHBOR_METADATA' in config_facts else {}
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+    nbr_port_map = mg_facts['minigraph_port_name_to_alias_map'] \
+        if 'minigraph_port_name_to_alias_map' in mg_facts else {}
+    for neighbor in neighs:
+        local_ip = neighs[neighbor]['local_addr']
+        nbr_port = get_port_by_ip(config_facts, local_ip)
```
Comment (deepak-singhal0408) [inline tests/route/test_route_flap.py:120]: nbr_port could be Ethernet<> or PortChannel<>.. right?
how are you handling portchannel case here?

Comment (sanjair-git) [reply tests/route/test_route_flap.py:120]: '_dev_port_' which is passed to this method '_get_neighbor_info_' either can be Portchannel<> or Ethernet<>.

And __get_port_by_ip__ method above returns the nbr_port based on the dut config_facts of '_PORTCHANNEL_INTERFACE_' and '_INTERFACE_'. So, if the nbr_port returned by _get_port_by_ip_ method is a Portchannel<> which can be directly same as '_dev_port_' passed through '_get_neighbor_info_'. Using that, we are fetching the neighbor's name and its type.

But if the nbr_port returned by 'get_port_by_ip' is Ethernet<>, it is actually **alias interface** name, not the actual interface name. So, we get the actual interface name using the _port_name_to_alias_map_ from minigraph facts for fetching the neighbor's name and its type.

**Comment by sanjair-git** [comment]
**Before the PR change:**
 
Test fetches only the ptf receiving ports of the DUT under test. T1 neighbor routes are same in all downstream DUTs. ping packet gets forwarded to other T1 neighbor DUT's ports instead of the DUT ports under test.
 
**After the PR change:**
 
Test fetches all the ptf receiving ports of all of the DUTs which has same type of neighbors similar to dut under test. Due to this change, we are making sure ping packet is received on any of the DUTs' ptf ports having similar neighbors (T1/T3).

File: tests/route/test_route_flap.py (hunk 1 above)
Comment (deepak-singhal0408) [reply tests/route/test_route_flap.py:120]: got it. thanks for the explanation..

**Comment by deepak-singhal0408** [comment]
LGTM

**Review by deepak-singhal0408 — APPROVED** [review]

**Comment by deepak-singhal0408** [comment]
@judyjoseph the changes look good to me.. Please help with the merge.

**Review by rlhui — APPROVED** [review]

**Comment by rajendrat** [comment]
viz @abdosi : Please help merge this to 202305 branch.
@vperumal

**Comment by sanjair-git** [comment]
> @sanjair-git please create PR for 202205/202305 as their is conflict in cherry-pick.

@abdosi , For both 202205/202305, cherry-pick PRs have been created already. Please see below,
202205: https://github.com/sonic-net/sonic-mgmt/pull/12137
202305: https://github.com/sonic-net/sonic-mgmt/pull/12138

**Pull request merged**